src/Script.cpp

```diff
@@ -698,7 +698,17 @@ int Script::_globalUnpersist(lua_State *L) {
                          loader.version(), DAGON_VERSION_STRING);
   }
 
+  // Create new room from file. Remove existing room first.
+  lua_getglobal(L, loader.roomName().c_str());
+  Room *oldRoom = ProxyToRoom(L, -1);
+  lua_pop(L, 1); // Pop Room.
+
   Script::instance()._loadRoomFile(L, loader.roomName().c_str());
+
+  if (oldRoom)
+    Control::instance().replaceRoom(oldRoom);
```

Comment:
Is this really okay @agustincordes ?
It feels very improper to just remove the pointer from the vector.